mlir/include/mlir/Conversion/GENToLLVM/GenIntrinsicEnum.h

```diff
@@ -0,0 +1,399 @@
+//===- GenIntrinsicEnum.h - IGC generated intrinsic declaration--*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This provides GenISAIntrinsic Enum definitions for GEN dialect use
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "llvm/IR/Intrinsics.h"
+
+#include <stdint.h>
+
+namespace llvm {
+
+namespace GenISAIntrinsic {
+
+enum ID : uint32_t {

```

Comment:
Could this be an `enum class`? 

Do we need all the entries for the current dialect? Or do we define them all to match some intrinsic definition in IGC?